rustler_codegen/src/nif.rs

```diff
@@ -15,16 +15,14 @@ pub fn transcoder_decorator(args: syn::AttributeArgs, fun: syn::ItemFn) -> Token
     let arity = arity(inputs.clone());
     let decoded_terms = extract_inputs(inputs.clone());
     let argument_names = create_function_params(inputs.clone());
-    let erl_func_name = extract_attr_value(args, "name")
-        .map(|ref n| syn::Ident::new(n, Span::call_site()))
-        .unwrap_or_else(|| name.clone());
+    let erl_func_name = extract_attr_value(args, "name").unwrap_or_else(|| name.to_string());
```

Comment:
From looking at the code, the function name has to be Latin1 encoded (and that also limits the allowed characters).

Reply:
I'm not saying using unicode isn't cumbersome, but since functions and variables in Elixir are just atoms, and atoms allow anything strings do, you can do
```elixir
defmodule Test do
  @cheese_var Macro.var(:"🧀s", __MODULE__)
  @doc """
  Hex needs cheese for the mouse
  """
  def unquote(:"🧀")(unquote(@cheese_var)) do
    IO.inspect(unquote(@cheese_var), label: "Trying to fetch cheese #")

    if unquote(@cheese_var) > 2 do
      {:error, "Out of cheese error! Redo from start."}
    else
      :ok
    end
  end
end

IO.inspect(apply(Test, :"🧀", [1]), label: "Getting 🧀")
```

Reply:
Did you test this? I checked the `erl_nif.c` and it explicitly uses latin1 atoms as far as I can see.

Reply:
I've added a check for it now :)